### PR TITLE
Monthly Normalization

### PR DESCRIPTION
**Pull request by thatzopoulos**
Issue : https://github.com/timescale/timescaledb-toolkit/issues/466

Added two new sql functions :
- days_in_month
- month_normalize

### CODE REVIEW

**Comment by thatzopoulos** [comment]
All changes in `utilities.rs` before line 88 are from `cargo fmt` and can be removed if necessary

File: extension/src/utilities.rs [hunk 1]
```diff
@@ -53,26 +61,47 @@ pub fn generate_periodic_normal_series(
...
+create or replace function toolkit_experimental.days_in_month(date timestamptz) RETURNS int AS $$\n\
+SELECT CAST(EXTRACT('day' FROM ($1 + interval '1 month' - $1)) as INTEGER)\n\
+$$ LANGUAGE SQL STRICT IMMUTABLE PARALLEL SAFE;\n\
+",
+    name = "days_in_month",
+);
+
+// Normalizes metric based on reference date and days
+extension_sql!("\n\
+create or replace function toolkit_experimental.month_normalize(metric float8, reference_date timestamptz, days float8 DEFAULT 365.25/12) RETURNS float8 AS $$\n\
```
Comment (davidkohn88) [inline extension/src/utilities.rs:100]: @JLockerman Do we need to do something for these in terms dealing with the potential security vulnerability of using CREATE OR REPLACE?

File: extension/src/utilities.rs [hunk 2]
```diff
@@ -110,4 +149,28 @@ mod tests {
...
+        });
+    }
+    #[pg_test]
+    fn test_monthly_normalize() {
+        Spi::execute(|client| {
+            let test_val = client.select("SELECT toolkit_experimental.month_normalize(1000,'2021-01-01 00:00:00+03'::timestamptz)",None,None,).first().get_one::<f64>().unwrap();
+            assert_eq!(test_val, 981.8548387096774f64);
+        });
+        Spi::execute(|client| {
+            let test_val = client.select("SELECT toolkit_experimental.month_normalize(1000,'2021-01-01 00:00:00+03'::timestamptz,30.5)",None,None,).first().get_one::<f64>().unwrap();
```
Comment (davidkohn88) [inline extension/src/utilities.rs:172]: I'd want to test with a value like 30 to make sure we don't accidentally do integer arithmetic on this. (Could just add one more test case).

**Review by davidkohn88 — COMMENTED** [review]
Only big thing is the potential security issue with using create or replace, I wonder how we can handle that and if we need to do something about it for `time_bucket_range` as well.

File: extension/src/utilities.rs (hunk 1 above)
Comment (JLockerman) [reply extension/src/utilities.rs:100]: @davidkohn88 if it's in experimental I think we can do an unconditional `CREATE`; the experimental schema is always dropped on update so any collision is a bug.

Comment (davidkohn88) [reply extension/src/utilities.rs:100]: What about when we graduate it?

Comment (davidkohn88) [reply extension/src/utilities.rs:100]: But for now that should work...

**Review by davidkohn88 — APPROVED** [review]
LGTM

**Review by davidkohn88 — APPROVED** [review]

**Pull request merged**